src/TodoContext.tsx

```diff
@@ -0,0 +1,317 @@
+}
+
+export interface TodoContextValue {
+  todos: Todo[];
+  postTodo: string;
+  filteredTodos: Todo[];
+  filter: string;
+  error: string;
+  existingCompleted: boolean;
+  nonCompletedTodos: number;
+  disableInput: boolean
+  isLoading: number[];
+  titleField: React.MutableRefObject<HTMLInputElement>;
+  tempTodo: Todo | null;
+  isChosenToRename: number;
+  editingTodo: string;
+  handleEditing: number;
+  setHandleEditing: (id: number) => void;
+  setEditingTodo: (qury: string) => void;
+  setIsLoading: React.Dispatch<React.SetStateAction<number[]>>;
+  setError: (errorMessage: string) => void;
+  setFilter: (newFilter: string) => void;
+  handleSubmit: () => void;
+  setPostTodo: (postTodo: string) => void;
+  setFilteredTodos: (todos: Todo[]) => void;
+  setDisableInput: (bollean: boolean) => void;
+  handleDelete: (id: number) => void;
+  handleCompletedDelete: () => void,
+  makeTodoCompleted: (id: number, isCompleted: boolean) => void,
+  setIsChosenToRename: (id: number) => void,
+  makeTodoChange: (id: number, value: string) => void,
+}
+
+export const TodoContext = createContext<TodoContextValue>({
+  todos: [],
+  postTodo: '',
+  filteredTodos: [],
+  filter: '',
+  error: '',
+  existingCompleted: false,
+  nonCompletedTodos: 0,
+  disableInput: false,
+  isLoading: [],
+  tempTodo: null,
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  titleField: null!,
+  isChosenToRename: 0,
+  editingTodo: '',
+  handleEditing: 0,
+  setEditingTodo: () => {},
+  setIsLoading: () => {},
+  setError: () => {},
+  setFilter: () => {},
+  handleSubmit: () => {},
+  setPostTodo: () => {},
+  setFilteredTodos: () => {},
+  setDisableInput: () => {},
+  handleDelete: () => {},
+  handleCompletedDelete: () => {},
+  makeTodoCompleted: () => {},
+  setIsChosenToRename: () => {},
+  makeTodoChange: () => {},
+  setHandleEditing: () => {},
+});
+
+const USER_ID = 104;
+
+export const TodoProvider: React.FC<TodoContextProps> = ({ children }) => {
+  const [postTodo, setPostTodo] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState('all');
+  const [error, setError] = useState('');
+  const [disableInput, setDisableInput] = useState(false);
+  const [isLoading, setIsLoading] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  // const [doubleClickCounter, setDoubleClickCounter] = useState(0);
+  const [isChosenToRename, setIsChosenToRename] = useState(0);
+  const [editingTodo, setEditingTodo] = useState('');
+  const [handleEditing, setHandleEditing] = useState(0);
+  const [timeToFocus, setTimeToFocus] = useState(0);
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const titleField = useRef<HTMLInputElement>(null!);
+
+  useEffect(() => {
+    if (titleField?.current) {
+      titleField.current.focus();
+    }
+  }, [timeToFocus]);
```

Comment:
you use ` titleField.current.focus()` too many times, also you don't need to create `timeToFocus`

what if we try to do the next:
```
useEffect(() => {
    if (titleField.current) {
      titleField.current.focus();
    }
  }, [tempTodo, todos.length]);
```

and remove all your `titleField.current.focus();` in other places